### PR TITLE
Make date format localizable after "this post was submitted on"

### PR DESCRIPTION
**Pull request by noamraph**
Current translation doesn't affect thing.datefmt (it's always "%d %b %Y"), so I changed it so that the format string is translated immediately.

### CODE REVIEW

File: r2/r2/templates/linkinfobar.html [hunk 1]
```diff
@@ -39,7 +39,7 @@
       &#32;
     </span>
     <time datetime="${html_datetime(thing.a._date)}">
-      ${thing.a._date.strftime(thing.datefmt)}
+      ${thing.a._date.strftime(_('%d %b %Y'))}
```
Comment (bsimpson63) [inline r2/r2/templates/linkinfobar.html]: something from babel might be more appropriate http://babel.pocoo.org/docs/api/dates/

Comment (noamraph) [inline r2/r2/templates/linkinfobar.html]: Do you mean something like

    format_date(thing.a._date, format=_('dd MMM yyyy'), locale=_('en_US'))

How do I get the locale name?

Comment (spladug) [inline r2/r2/templates/linkinfobar.html]: `c.locale` should contain the current locale. See [other users of the babel functions for examples](https://github.com/reddit/reddit/blob/master/r2/r2/lib/template_helpers.py#L586-L590).

File: r2/r2/lib/template_helpers.py [hunk 2]
```diff
@@ -592,6 +593,11 @@ def format_percent(ratio, locale=None):
     return babel.numbers.format_percent(ratio, locale=locale or c.locale)
 
 
+def format_date(date, locale=None):
+    return babel.dates.format_date(date, _("dd MMM yyyy"),
```
Comment (bsimpson63) [inline r2/r2/lib/template_helpers.py]: Why is the format string marked for translation?

**Comment by umbrae** [comment]
Closing due to idleness, feel free to reopen if you intend to continue working on it.

**Pull request closed without merging**